undefined reference to `llvm::SampleContextTracker::getContextString[abi:cxx11](llvm::sampleprof::FunctionSamples const&) const'

When building ldc on Fedora, I get the following error:

```
[ 50%] Linking CXX executable ../../../bin/ldc-profgen
/usr/bin/ld: CMakeFiles/ldc-profgen.dir/CSPreInliner.cpp.o: in function `llvm::sampleprof::CSPreInliner::processFunction(llvm::StringRef)':
CSPreInliner.cpp:(.text+0xbad): undefined reference to `llvm::SampleContextTracker::getContextString[abi:cxx11](llvm::sampleprof::FunctionSamples const&) const'
/usr/bin/ld: CSPreInliner.cpp:(.text+0xe0a): undefined reference to `llvm::SampleContextTracker::getContextString[abi:cxx11](llvm::sampleprof::FunctionSamples const&) const'
/usr/bin/ld: CMakeFiles/ldc-profgen.dir/CSPreInliner.cpp.o: in function `llvm::sampleprof::CSPreInliner::run()::{lambda(llvm::SampleContextTracker&, bool)#1}::operator()(llvm::SampleContextTracker&, bool) const':
CSPreInliner.cpp:(.text+0xf8c): undefined reference to `llvm::SampleContextTracker::getContextString[abi:cxx11](llvm::ContextTrieNode*) const'
collect2: error: ld returned 1 exit status
make[2]: *** [tools/ldc-profgen/ldc-profgen-17.0/CMakeFiles/ldc-profgen.dir/build.make:177: bin/ldc-profgen] Error 1
make[1]: *** [CMakeFiles/Makefile2:555: tools/ldc-profgen/ldc-profgen-17.0/CMakeFiles/ldc-profgen.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```

I get the exact same error with both 1.36.0 release and also with latest git master (that had [2.107](https://github.com/ldc-developers/ldc/commit/eeeebff0a9aa2b7b11072e59e49b17d69904f175) just merged).

It looks like something is going wrong with building against llvm that has debug disabled. CSPreInliner::run() uses NDEBUG to guard the debug code and apparently NDEBUG isn't passed:

```
[ 50%] Building CXX object tools/ldc-profgen/ldc-profgen-17.0/CMakeFiles/ldc-profgen.dir/CSPreInliner.cpp.o
cd /home/kalev/tmp/ldc/build/tools/ldc-profgen/ldc-profgen-17.0 && /usr/lib64/ccache/c++ -DLDC_ENABLE_PLUGINS -DLDC_LLVM_SUPPORTED_TARGET_AArch64=1 -DLDC_LLVM_SUPPORTED_TARGET_AMDGPU=1 -DLDC_LLVM_SUPPORTED_TARGET_ARM=1 -DLDC_LLVM_SUPPORTED_TARGET_AVR=1 -DLDC_LLVM_SUPPORTED_TARGET_BPF=1 -DLDC_LLVM_SUPPORTED_TARGET_Hexagon=1 -DLDC_LLVM_SUPPORTED_TARGET_Lanai=1 -DLDC_LLVM_SUPPORTED_TARGET_LoongArch=1 -DLDC_LLVM_SUPPORTED_TARGET_MSP430=1 -DLDC_LLVM_SUPPORTED_TARGET_Mips=1 -DLDC_LLVM_SUPPORTED_TARGET_NVPTX=1 -DLDC_LLVM_SUPPORTED_TARGET_PowerPC=1 -DLDC_LLVM_SUPPORTED_TARGET_RISCV=1 -DLDC_LLVM_SUPPORTED_TARGET_Sparc=1 -DLDC_LLVM_SUPPORTED_TARGET_SystemZ=1 -DLDC_LLVM_SUPPORTED_TARGET_VE=1 -DLDC_LLVM_SUPPORTED_TARGET_WebAssembly=1 -DLDC_LLVM_SUPPORTED_TARGET_X86=1 -DLDC_LLVM_SUPPORTED_TARGET_XCore=1 -I/home/kalev/tmp/ldc/. -I/home/kalev/tmp/ldc/dmd -DDMDV2 -I/usr/include -std=c++17   -fno-exceptions -funwind-tables -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -fno-rtti  -Wall -Wextra -Wno-unused-parameter -Wno-comment -Wno-missing-field-initializers -Wno-non-virtual-dtor -Wno-pedantic -DLDC_POSIX  -DIN_LLVM -DOPAQUE_VTBLS "-DLDC_INSTALL_PREFIX=R\"(/usr/local)\"" -DLDC_LLVM_VER=1700 "-DLDC_LIBDIR_SUFFIX=R\"()\"" -DLDC_HOST_LDMD=1 -DLDC_HOST_FE_VER=2103 "-DLDC_LLVM_LIBDIR=R\"(/usr/lib64)\"" -MD -MT tools/ldc-profgen/ldc-profgen-17.0/CMakeFiles/ldc-profgen.dir/CSPreInliner.cpp.o -MF CMakeFiles/ldc-profgen.dir/CSPreInliner.cpp.o.d -o CMakeFiles/ldc-profgen.dir/CSPreInliner.cpp.o -c /home/kalev/tmp/ldc/tools/ldc-profgen/ldc-profgen-17.0/CSPreInliner.cpp
```

(There's no NDEBUG on the line above.)

As a side note, I wonder if it would make sense to add Fedora to CI, using Fedora's llvm and the ldc version that's part of the distro? This would have helped to catch this earlier.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -505,13 +505,13 @@ message(STATUS "-- Building LDC with integrated LLD linker (LDC_WITH_LLD): ${LDC
 
 message(STATUS "-- Building LDC with enabled assertions (LDC_ENABLE_ASSERTIONS): ${LDC_ENABLE_ASSERTIONS}")
 if(LDC_ENABLE_ASSERTIONS)
-    append("-UNDEBUG" EXTRA_CXXFLAGS)
+    append("-UNDEBUG" LDC_CXXFLAGS)
     # avoid MSVC warning D9025 about "-DNDEBUG ... -UNDEBUG"
     string(REGEX REPLACE "(^| )[/-]D *NDEBUG( |$)" "\\1-UNDEBUG\\2" CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
     string(REGEX REPLACE "(^| )[/-]D *NDEBUG( |$)" "\\1-UNDEBUG\\2" CMAKE_CXX_FLAGS_MINSIZEREL "${CMAKE_CXX_FLAGS_MINSIZEREL}")
     string(REGEX REPLACE "(^| )[/-]D *NDEBUG( |$)" "\\1-UNDEBUG\\2" CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO}")
 else()
-    append("-DNDEBUG" EXTRA_CXXFLAGS)
+    append("-DNDEBUG" LDC_CXXFLAGS)
 endif()
 
 #

